RecoBTag/ONNXRuntime/plugins/DeepCombinedONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,345 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/BTauReco/interface/DeepFlavourTagInfo.h"
+
+#include "PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h"
+
+#include "RecoBTag/ONNXRuntime/interface/tensor_fillers.h"
+
+using namespace cms::Ort;
+
+class DeepCombinedONNXJetTagsProducer : public edm::stream::EDProducer<edm::GlobalCache<ONNXRuntime>> {
+public:
+  explicit DeepCombinedONNXJetTagsProducer(const edm::ParameterSet&, const ONNXRuntime*);
+  ~DeepCombinedONNXJetTagsProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+  static std::unique_ptr<ONNXRuntime> initializeGlobalCache(const edm::ParameterSet&);
+  static void globalEndJob(const ONNXRuntime*);
+
+private:
+  typedef std::vector<reco::DeepFlavourTagInfo> TagInfoCollection;
+  typedef reco::JetTagCollection JetTagCollection;
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  void make_inputs(unsigned i_jet, const reco::DeepFlavourTagInfo& taginfo);
+
+  const edm::EDGetTokenT<TagInfoCollection> src_;
+  std::vector<std::string> flav_names_;
+  std::vector<std::string> input_names_;
+  std::vector<std::string> output_names_;
+
+  const double min_jet_pt_;
+  const double max_jet_eta_;
+
+  enum InputIndexes {
+    kGlobal = 0,
+    kChargedCandidates = 1,
+    kNeutralCandidates = 2,
+    kVertices = 3,
+    kGlobal1 = 4,
+    kSeedingTracks = 5,
+    kNeighbourTracks = 6
+  };
+  constexpr static unsigned n_features_global_ = 15;
+  constexpr static unsigned n_cpf_ = 25;
+  constexpr static unsigned n_features_cpf_ = 16;
+  constexpr static unsigned n_npf_ = 25;
+  constexpr static unsigned n_features_npf_ = 6;
+  constexpr static unsigned n_sv_ = 4;
+  constexpr static unsigned n_features_sv_ = 12;
```

Comment:
Put these in a header and import here as well as in DeepVertex. A follow-up PR can clean up DeepFlavour to include them directly there as well, rather than specifying multiple times.

Reply:
just to be clear, you would put in the const expressions in config header and then call them in the producer by name instead of the raw number?

Reply:
Yes, then you can use the same definitions in all the producers that you introduce here, and eventually also in DeepFlavour.